django/core/mail/backends/filebased.py

```diff
@@ -17,9 +17,6 @@ def __init__(self, *args, file_path=None, **kwargs):
             self.file_path = file_path
         else:
             self.file_path = getattr(settings, 'EMAIL_FILE_PATH', None)
-        # Make sure self.file_path is a string.
-        if not isinstance(self.file_path, str):
```

Comment:
Reinstate this, but add `pathlib.Path` to the `isinstance` check. The comment was staying the obvious and can go.

Reply:
I had considered this direction on the first approach but chose to remove the type checking instead. I feel that this better follows Python's duck typing tradition. The value of `self.file_path` is always passed to `os.path.abspath()` anyway which does its own type checking, so Django doesn't need to duplicate it. So invalid values are still rejected.

I've added a test for the `os.path.abspath()` type checking.